test/integration/utils/assetPricing.js

```diff
@@ -25,15 +25,29 @@ async function assetToEthWithPrecisionLoss(pool, coverAmountInAsset, assetToEthR
 
   return coverAmountInNXM.mul(nxmEthPrice).div(config.ONE_NXM);
 }
+async function assetToNXM(pool, amountInAsset, assetID, config) {
+  const nxmPriceInCoverAsset = BigNumber.from(await pool.getTokenPriceInAsset(assetID));
+  const assetWithDecimals = BigNumber.from(amountInAsset).mul(config.ONE_NXM);
+  const amountInNXMRaw = divCeil(assetWithDecimals, nxmPriceInCoverAsset);
+  const amountInNXM = roundUpToNearestAllocationUnit(amountInNXMRaw, config.NXM_PER_ALLOCATION_UNIT);
+  return amountInNXM;
+}
+
+async function NXMToAsset(pool, amountInNXM, assetID, config) {
+  const nxmPriceInCoverAsset = BigNumber.from(await pool.getTokenPriceInAsset(assetID));
+  const nxmWithDecimals = BigNumber.from(amountInNXM).mul(config.ONE_NXM);
+  const amountInAssetRaw = divCeil(nxmWithDecimals, nxmPriceInCoverAsset);
+  const amountInAsset = roundUpToNearestAllocationUnit(amountInAssetRaw, config.NXM_PER_ALLOCATION_UNIT);
+  return amountInAsset;
+}
 
-// Replicates the amount stored when buying cover with asset other than NXM
+// Converts amount in asset to NXM and back to asset with precision loss
 async function assetWithPrecisionLoss(pool, amountInAsset, assetID, config) {
-  const nxmPriceInCoverAsset = await pool.getTokenPriceInAsset(assetID);
-  const amountInNXM = roundUpToNearestAllocationUnit(
-    divCeil(amountInAsset.mul(config.ONE_NXM), nxmPriceInCoverAsset),
-    config.NXM_PER_ALLOCATION_UNIT,
-  );
+  const nxmPriceInCoverAsset = BigNumber.from(await pool.getTokenPriceInAsset(assetID));
+  const nxmRoundedUp = BigNumber.from(amountInAsset).mul(config.ONE_NXM);
+  const amountInNXMRaw = divCeil(nxmRoundedUp, nxmPriceInCoverAsset);
+  const amountInNXM = roundUpToNearestAllocationUnit(amountInNXMRaw, config.NXM_PER_ALLOCATION_UNIT);
```

Comment:
These functions don't do what they say they do. Nxm <> asset conversion and allocation units are different things. Generally avoid adding any helper functions that do more than one thing. Helper functions should be *generic* (think linux commands - do a single thing and do it well). I would actually avoid having per-folder helper functions and would keep everything inside the test file, close to the test itself, otherwise we'll start adding TONS of functions (and we started already), they never do what they say - you always have to navigate to the definition to see what's actually happening. This is opaque and error prone.

Reply:
And you seem to use this function in A SINGLE PLACE....

Reply:
Moved the discussion to chat.